pyccel/codegen/printing/fcode.py

```diff
@@ -705,7 +878,29 @@ def _print_NumpyRandint(self, expr):
         if expr.rank != 0:
             errors.report(FORTRAN_ALLOCATABLE_IN_EXPRESSION,
                           symbol=expr, severity='fatal')
-        return expr.fprint(self._print)
+        if expr.high is None:
+            randreal = self._print(PyccelMul(expr.low, NumpyRand()))
+        else:
+            randreal = self._print(PyccelAdd(PyccelMul(PyccelAssociativeParenthesis(PyccelMinus(expr.high, expr.low)), NumpyRand()), expr.low))
+
+        prec_code = self._print(expr.precision)
+        return 'floor({}, kind={})'.format(randreal, prec_code)
+    
+    def _print_NumpyFull(self, expr, lhs_code):
```

Comment:
what about the if statements here?

Reply:
The `if` statement is included to handle the case of `NumpyEmpty`. I think there should be an `if` statement in the `Assign` to handle this case.

Reply:
You will need to make one additional modification to deal with stupid users ;) . If a user does:
```
a = np.empty(3)*5
```
Then [this error](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2574-L2575) should be raised. This can be done by adding a specialised `_print_NumpyEmpty` function now that `NumpyOnes` and `NumpyZeros` no longer inherit from `NumpyEmpty`. You could modify `tests/errors/codegen/fortran/FORTRAN_ALLOCATABLE_IN_EXPRESSION.py` to test this

Reply:
should we still need a specialized `_print_NumpyEmpty`after we're working with:
`if isinstance(rhs, NumpyEmpty):
            return ''`
for `tests/errors/codegen/fortran/FORTRAN_ALLOCATABLE_IN_EXPRESSION.py` I removed it ;).

Reply:
Probably we do need a specialized `_print_NumpyEmpty` to make sure the error is raised. If you do:
```
a = np.empty(3)*5
```
Then you have:
```
PyccelAssign(Variable('a'), PyccelMul(NumpyEmpty(LiteralInteger(3)), LiteralInteger(5))
```
So you won't reach the if block `if isinstance(rhs, NumpyEmpty): return ''`

Reply:
> for `tests/errors/codegen/fortran/FORTRAN_ALLOCATABLE_IN_EXPRESSION.py` I removed it ;).

You can get it back by doing:
```
git checkout master -- tests/errors/codegen/fortran/FORTRAN_ALLOCATABLE_IN_EXPRESSION.py
```